src/Microsoft.AspNetCore.Authentication.Facebook/FacebookMiddleware.cs

```diff
@@ -22,20 +22,21 @@ public class FacebookMiddleware : OAuthMiddleware<FacebookOptions>
         /// Initializes a new <see cref="FacebookMiddleware"/>.
         /// </summary>
         /// <param name="next">The next middleware in the HTTP pipeline to invoke.</param>
+        /// <param name="serviceProvider"></param>
         /// <param name="dataProtectionProvider"></param>
         /// <param name="loggerFactory"></param>
         /// <param name="encoder"></param>
         /// <param name="sharedOptions"></param>
         /// <param name="options">Configuration options for the middleware.</param>
-        /// <param name="configureOptions"></param>
         public FacebookMiddleware(
             RequestDelegate next,
+            IServiceProvider serviceProvider,
             IDataProtectionProvider dataProtectionProvider,
             ILoggerFactory loggerFactory,
             UrlEncoder encoder,
             IOptions<SharedAuthenticationOptions> sharedOptions,
             IOptions<FacebookOptions> options)
-            : base(next, dataProtectionProvider, loggerFactory, encoder, sharedOptions, options)
+            : base(next, serviceProvider, dataProtectionProvider, loggerFactory, encoder, sharedOptions, options)
```

Comment:
Instead of doing service locator for the ITokenStore, maybe we should just stick a NoOpTokenStore into DI via TryAdd by default, and always call into the store.